File: samples/Test/Classes/Control/StepperTest.cpp

#include "StepperTest.h"

StepperTest::StepperTest()
{
    this->setTitle("CAStepper");
}

StepperTest::~StepperTest()
{
}

void StepperTest::viewDidLoad()
{
    CALabel* stepValue = CALabel::createWithLayout(DLayout(DHorizontalLayout_W_C(300, 0.5), DVerticalLayout_H_C(100, 0.15)));
    stepValue->setText("step_value:0");
    stepValue->setColor(CAColor4B::BLACK);
    stepValue->setFontSize(28);
    stepValue->setTextAlignment(CATextAlignment::Center);
    stepValue->setVerticalTextAlignmet(CAVerticalTextAlignment::Center);
    this->getView()->addSubview(stepValue);
    
    CAStepper* stepper = CAStepper::createWithLayout(DLayout(DHorizontalLayout_W_C(360, 0.5), DVerticalLayout_H_C(60, 0.3)), CAStepper::Orientation::Horizontal);
    //stepper->setWraps(true);//是否循环,默认循环
    stepper->setMinValue(0);
    stepper->setMaxValue(50);
    stepper->setStepValue(1);
    //stepper->setAutoRepeat(false);
    this->getView()->addSubview(stepper);
    stepper->setTarget([=](float value){
        
        char tem[30];
        sprintf(tem, "step_value:%.0f",value);
        stepValue->setText(tem);
        CCLog("step-tag === %f",value);
    });
    
    CALabel* stepValue1 = CALabel::createWithLayout(DLayout(DHorizontalLayout_W_C(200, 0.5), DVerticalLayout_H_C(100, 0.6)));
    stepValue1->setText("step_value:0");
    stepValue1->setColor(CAColor4B::BLACK);
    stepValue1->setFontSize(28);
    stepValue1->setTextAlignment(CATextAlignment::Center);
    stepValue1->setVerticalTextAlignmet(CAVerticalTextAlignment::Center);
    this->getView()->addSubview(stepValue1);
    
    CAStepper* stepper1 = CAStepper::createWithLayout(DLayout(DHorizontalLayout_W_C(60, 0.5), DVerticalLayout_H_C(200, 0.8)), CAStepper::Orientation::Vertical);
    //stepper1->setWraps(true);//是否循环,默认循环
    stepper1->setMinValue(0);
    stepper1->setMaxValue(50);
    stepper1->setStepValue(1);
    //stepper1->setAutoRepeat(false);
    this->getView()->addSubview(stepper1);
    stepper1->setTarget([=](float value){
        
        char tem[30];
        sprintf(tem, "step_value:%.0f",value);
        stepValue1->setText(tem);
        CCLog("step-tag === %f",value);
    });
}

void StepperTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/StepperTest.js
/**
 * Created by crossApp on 16/8/3.
 */
var StepperTest = ca.CAViewController.extend({

    ctor: function () {
        this._super();
    },

    viewDidLoad: function() {

        this.step_value = ca.CALabel.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(250, 0.5), ca.DVerticalLayout_H_C(100 , 0.15)));
        this.step_value.setText("value = 0");
        this.step_value.setFontSize(28);
        this.step_value.setTextAlignment(ca.CATextAlignment.Center);
        this.step_value.setVerticalTextAlignmet(ca.CAVerticalTextAlignment.Center);
        this.getView().addSubview(this.step_value);

        var step = ca.CAStepper.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(240, 0.5), ca.DVerticalLayout_H_C(100. , 0.3)), ca.CAStepper.Orientation.Horizontal);
        step.setMinValue(0);
        step.setMaxValue(50);
        step.setStepValue(1);
        //step.setAutoRepeat(false);
        this.getView().addSubview(step);
        var valueLabel = this.step_value;
        step.setTarget(function (value) {
            log("stepperCallBack-->");
            var str = "value = " + value;
            valueLabel.setText(str);
        });


        /********************************************************************************************/


        this.step_value2 = ca.CALabel.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(250, 0.5), ca.DVerticalLayout_H_C(100 , 0.65)));
        this.step_value2.setText("value = 0");
        this.step_value2.setFontSize(28);
        this.step_value2.setTextAlignment(ca.CATextAlignment.Center);
        this.step_value2.setVerticalTextAlignmet(ca.CAVerticalTextAlignment.Center);
        this.getView().addSubview(this.step_value2);


        var step2 = ca.CAStepper.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(60, 0.5), ca.DVerticalLayout_H_C(200 , 0.8)), ca.CAStepper.Orientation.Vertical);
        step2.setMinValue(0);
        step2.setMaxValue(50);
        step2.setStepValue(1);
        this.getView().addSubview(step2);
        var valueLabel2 = this.step_value2;
        step2.setTarget(function (value) {
            log("stepperCallBack-->");
            var str = "value = " + value;
            valueLabel2.setText(str);
        });

    },

});
